src/control/lib/hardware/hwprov/defaults.go

```diff
@@ -59,3 +61,15 @@ func DefaultFabricScanner(log logging.Logger) *hardware.FabricScanner {
 
 	return fs
 }
+
+// Init loads up any dynamic libraries that need to be loaded at runtime.
+func Init(log logging.Logger) (func(), error) {
```

Comment:
It feels like this function, which initialises specific libraries within lib/hardware, should be in a file within the base lib/hardware directory rather than a subpackage, something like lib/hardware/main.go so that when the lib/hardware package is imported the relevant libraries are loaded. Not a big issue, just an initial reaction.

Reply:
The reason we wound up creating the hwprov package is because lib/hardware is imported by all of the sub-packages so they can use its types. So lib/hardware can't reference the subpackages Itself without creating an import cycle. hwprov's job is to be the central point that deals with all the concrete subpackages, basically.